docs/faq.rst

```diff
@@ -291,6 +292,70 @@ See :ref:`encrypted_repos` for more details.
 
+Using ``BORG_PASSCOMMAND`` with a properly permissioned file
+  Another option is to create a file with a password in it in your home
+  directory and use permissions to keep anyone else from reading it. For
+  example, first create a key::
+
+    dd if=/dev/urandom of=~/.borg-passphrase bs=1024 count=1
+    chmod 400 ~/.borg-passphrase
+
+  Then in an automated script one can put::
+
+    export BORG_PASSCOMMAND="cat ~/borg-passphrase"
+
+  and |project_name| will automatically use that passphrase.
+
+Using ``BORG_PASSCOMMAND`` with MacOS Keychain
+  MacOS has a native manager for secrets (such as passphrases) which is safer
+  than just using a file as it is encrypted at rest and unlocked manually
+  (fortunately, the login keyring automatically unlocks when you login). With
+  the built-in ``security`` command, you can access it from the command line,
+  making it useful for ``BORG_PASSCOMMAND``.
+
+  First generate a passphrase and use ``security`` to save it to your login
+  (default) keychain::
+
+    security add-generic-password -D secret -U -a $USER -s borg-passphrase -w $(head -c 30 /dev/urandom | xxd -p)
+
+  In your backup script retrieve it in the ``BORG_PASSCOMMAND``::
+
+    export BORG_PASSCOMMAND="security find-generic-password -a $USER -s borg-passphrase"
+
+Using ``BORG_PASSCOMMAND`` with GNOME Keyring (Ubuntu, Debian, Fedora)
+  GNOME has a keyring daemon similar to that of MacOS. However, it requires a bit more setup
+  to get the login keyring to work. First ensure ``libpam-gnome-keyring`` is installed and
+  running on login (skip this step on Fedora, as it is installed by default)::
+
+    sudo apt install gnome-keyring libpam-gnome-keyring
+    # the following commands may also be necessary if the previous command installed new packages
+    sudo sh -c "echo session optional pam_gnome_keyring.so auto_start >> /etc/pam.d/login"
+    sudo sh -c "echo password optional pam_gnome_keyring.so >> /etc/pam.d/passwd"
+    # you may need to relogin afterwards to activate the login keyring
+
+  Then add a secret to the login keyring::
+
+    sudo apt install libsecret-tools  # Ubuntu, Debian
+    sudo yum install libsecret-tools  # Fedora, RHEL
+    head -c 32 /dev/urandom | secret-tool store borg-repository repo-name --label="Borg Passphrase"
+
+  If a dialog box pops up prompting you to pick a password for a new keychain, use your
+  login password. If there is a checkbox for automatically unlocking on login, check it.
+
+  Once the secret is saved, retrieve it in a backup script using ``BORG_PASSCOMMAND``::
+
+    export BORG_PASSCOMMAND="secret-tool lookup borg-repository repo-name"
+
+  .. note:: For this to automatically unlock the keychain it must be run
+    in the ``dbus`` session of an unlocked terminal; for example, running a backup
+    script as a ``cron`` job might not work unless you also ``export DISPLAY=:0``
+    so ``secret-tool`` can pick up your open session.
```

Comment:
Note: accessing the session bus _of a different user_, even as root, was made more complex some time ago.

```
$ cat /etc/dbus-1/session-local.conf 
<busconfig>
<policy context="default">
<allow user="root"/>  <!-- clearly, root needs explicit permission
                           to do anything nowadays -->
</policy>
</busconfig>
```

env vars: DISPLAY, XAUTHORITY=/home/.../.Xauthority, DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus

Reply:
Not sure if this is the best place to comment, but I hope my this helps someone who stumbles of this and point me to the right place where it could be added.

This is how I use GNOME Keyring with borg in a user cron job (as I need a `unlocked  terminal` (which I read as unlocked GNOME session) anyway:
```
export BORG_PASSCOMMAND='sudo -u $USERNAME DISPLAY=:0 DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus secret-tool lookup borg-repository repo-name'
sudo -E borg create /path/to/repo:mybackup-{now} /path/to/backup 
````

The secret key was added as described as user `$USERNAME`:
```
secret-tool store borg-repository repo-name --label="Borg Passphrase"
```